pkg/status/status.go

```diff
@@ -88,6 +91,26 @@ func SendSyncMessage(err error) {
 	}
 }
 
+// ReportOperatorSourceError is used to report that an enabled default OperatorSource has encountered an error.
+func ReportOperatorSourceError(name string, err StatusError) {
+	// Do not report the error if the coAPI is not available or it is not an enabled default OperatorSource.
+	if instance == nil || instance.coAPINotPresent || !defaults.IsDefaultSource(name) {
+		return
+	}
+
+	instance.defaultOpsrcErrors[name] = err
+}
+
+// ReportOperatorSourceSuccess is used to report that an enabled default OperatorSource has entered the succeeded state.
+func ReportOperatorSourceSuccess(name string) {
+	// Do not report the success if the coAPI is not available or it is not an enabled default OperatorSource.
+	if instance == nil || instance.coAPINotPresent || !defaults.IsDefaultSource(name) {
+		return
+	}
+
+	delete(instance.defaultOpsrcErrors, name)
```

Comment:
Why could we not end up in a situation where we are concurrently trying to update the same global variable? And one is trying to delete it why the other is trying to add an entry?

Reply:
Given the purpose of this code it is unlikely that we will encounter a race conditio. The controller handles OperatorSource events linearly - so I felt that adding a lock was over engineering. @ecordell can weigh in.

Reply:
Even if this function is never called concurrently, we still need to be concerned about concurrent reads from `monitorClusterStatus`, right?

Reply:
That's true @ecordell, I'll look into adding a lock.